### ISSUE TITLE
deepscan ignores anyProjectDeps flag

### ISSUE THREAD

**Issue by fe3dback** [fe3dback/go-arch-lint#67]
config:
```yaml
deps:
  di:
    anyProjectDeps: true
    anyVendorDeps: true
```

with this configuration, linter return unexpected error:
```
Dependency x -\-> di not allowed
  ├─ x some.Client in /internal/generated/some/sdk.go:28
  └─ di New in /internal/container/container.go:43
 
     /project-path/main.go:258
        252 |   cont := container.New(
     >  258 |     app.someClient,
     
```

because of `anyProjectDeps`, all project dependencies must be allowed for `di`, but linter possibly ignores this flag for deepscan.

**Comment by fe3dback** [comment]
fixed in https://github.com/fe3dback/go-arch-lint/pull/78
